### ISSUE TITLE
Some thoughts on reproducibility - feel free to ignore/close 

### ISSUE THREAD

**Issue by seabbs** [informationcake/SDSS-ML#1]
Hi Alex, 

Nice paper + code!

I was just watching your video ahead of our group discussion and thought I would take a closer look at the code. All looks really good with some good comments and a concise readme. I just had a few comments about how it could have been a little easier to work through for me (Note: I am terrible at python and also am not a domain expert so perhaps I am barking up the wrong tree/trees).

## [Binder environment](https://mybinder.org) 

Not sure if you have come across it. It uses containers (docker) to spin up a remote cloud instance (though with limited compute) of your dev environment. Can be a really great way of getting people up to speed quickly with your code. It also forces you to write out the dependencies in a way that it can be installed from a script which is nice in general (though good job for actually listing them at the start of scripts!). 

Potential alternatives could be a dockerfile or anaconda environment (I think).

## Data

I see the data input lines in the code and the zenodo link to the data catalogue (which I think is the output from your code?) but I am not clear how to get the csv you reference in the code. I know you go through your data sources in the paper but I didn't have much luck here (I imagine this would all make more sense to domain expert!). I think the data is all publicly available but failing that it would be awesome to have synthetic dataset to run the analysis with. 

## Organisation of functions/scripts

Again this may be my own lack of knowledge but is it standard python practice to put the user defined functions ahead of the main code in each script (I guess this would be like C/C++). Personally I found it a little challenging to wade through the functions to get to the script section and would probably have preferred to have them in a separate file (which I could have looked at based on your nice comments at the top of each script). In R we typically package all the functions up and then call them into the script (with docs available using ?function). This approach might make it clearer how modular your code is and hence make it easier for people like me to borrow your nice functions for our own use. 

## Modelling 

Not really to do with the code and more general interest. Is there a reason you stuck to tuning a random forest model versus using a more (classically) high performing model like gradient boosting (xgboost, lightgbm etc.)? Extending beyond that, is there a reason that you didn't look at ensembling? In a similar vein it would be really interesting to see what kind of performance boost your approach gets over something a bit rubbish like a GLM. 

Sorry this is a bit long and feel free to ignore any/all of it!
